src/compiler/scala/tools/nsc/reporters/DelegatingReporter.scala

```diff
@@ -0,0 +1,41 @@
+package scala.tools.nsc
+package reporters
+
+import scala.reflect.internal.util.Position
+import scala.reflect.internal.settings.MutableSettings
+
+/** A `Reporter` that filters events before delegating.
+ *
+ *  Concrete subclasses should implement just the abstract `filter` method.
+ */
+abstract class DelegatingReporter(delegate: Reporter) extends Reporter {
+
+  /* True to permit the message. */
+  protected def filter(pos: Position, msg: String, severity: Severity): Boolean
```

Comment:
Should we move this method to `LimitingReporter` and make `DelegatingReporter` a pure delegate?

Reply:
OK, I didn't overthink. It was Filter before. I noticed the java tool convention is `Forwarding` as in `ForwardingJavaFileManager`. Thx for the second look.